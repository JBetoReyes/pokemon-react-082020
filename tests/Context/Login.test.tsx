import React from 'react';
import UserContext from '@components/hooks/useContext/UserContext.component';
import Login from '@components/hooks/useContext/Login.component';
import { mount } from 'enzyme';

describe('Login', () => {
  const user = {
    name: 'test name',
    email: 'test@gmail.com',
  };
  const state = {
    user,
    setUser: jest.fn(),
  };
  const loginJsx = (
    <UserContext.Provider value={state}>
      <Login />
    </UserContext.Provider>
  );
  let wrapper = mount(loginJsx);
  beforeEach(() => {
    wrapper = mount(loginJsx);
  });
  test('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
