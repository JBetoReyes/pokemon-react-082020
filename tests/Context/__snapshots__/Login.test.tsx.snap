// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should match the snapshot 1`] = `
<Login>
  <h1>
    Login component
  </h1>
  <hr />
  <button
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
</Login>
`;
